src/zope/interface/exceptions.py

```diff
@@ -97,19 +117,49 @@ class BrokenMethodImplementation(Invalid, _TargetMixin):
     """
 
     def __init__(self, method, mess, target=_NotGiven):
-        Invalid.__init__(self)
+        Invalid.__init__(self, method, mess, target)
         self.method = method
         self.mess = mess
         self.target = target
 
-    def __str__(self):
-        return "%s violates its contract in %s: %s." % (
-            self._prefix,
+    @property
+    def _specifics(self):
+        return 'violates the contract of %s because %s' % (
```

Comment:
... but this one doesn't.

How does one know, when overriding a `_specifics` property, whether it needs to start with a space or not?

Reply:
Because if one gets it wrong the tests break 😄 

I agree that's annoying. I'll see about unifying it. Thank you.